signingscript/sign.py

```diff
@@ -682,15 +704,86 @@ def _owner_filter(tarinfo_obj):
         raise SigningScriptError(e)
 
 
+async def call_autograph(url, user, password, request_json):
+    """Call autograph and return the json response."""
+    auth = HawkAuth(id=user, key=password)
+    with requests.Session() as session:
+        r = session.post(url, json=request_json, auth=auth)
+        log.debug("Autograph response: %s",
+                  r.text[:120] if len(r.text) >= 120 else r.text)
+        r.raise_for_status()
+        return r.json()
+
+
+def make_signing_req(input_bytes, server, fmt, keyid=None):
+    """Make a signing request object to pass to autograph."""
+    base64_input = base64.b64encode(input_bytes).decode('ascii')
+    sign_req = {"input": base64_input}
+
+    if keyid:
+        sign_req['keyid'] = keyid
+
+    # TODO: Is this the right place to do this?
```

Comment:
We may want to refactor or clean up signingscript when we're fully migrated to autograph. I think it's hard to move this logic in the current workflow.